Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp

```diff
@@ -108,6 +108,33 @@ ImageBufferSkiaAcceleratedBackend::~ImageBufferSkiaAcceleratedBackend()
 #endif
 }
 
+void ImageBufferSkiaAcceleratedBackend::finishAcceleratedRenderingAndCreateFence()
+{
+    auto* glContext = PlatformDisplay::sharedDisplay().skiaGLContext();
+    if (!glContext || !glContext->makeContextCurrent())
+        return;
+
+    auto* grContext = PlatformDisplay::sharedDisplay().skiaGrContext();
+    RELEASE_ASSERT(grContext);
+
+    if (GLFence::isSupported()) {
+        grContext->flushAndSubmit(m_surface.get(), GrSyncCpu::kNo);
+        m_fence = GLFence::create();
+        if (!m_fence)
+            grContext->submit(GrSyncCpu::kYes);
+    } else
+        grContext->flushAndSubmit(m_surface.get(), GrSyncCpu::kYes);
+}
```

Comment:
I do not know much about skia but I think this function is doing two separate things. Can't it be split into two functions: one flushes the surface to the shared context and the other creates the fence?
```
flushToSharedDisplay()
createFence()
```

Reply:
We've followed the current pattern already in CoordinatedTileBuffer, which contains literally the same code related to flushAndSubmit + fence creation -- I've tried to stick to that. But yes, in principle we could split it -- however, fence creation might fail, and then we need to call again submit, but with the GrSyncCpu::kYes option to make the CPU block here if the GPU driver doesn't support fencing.

Therefore I think it's better to combine flush/submit+fencing into one call, that contains the logic in a single place.